src/project-2.js

```diff
@@ -44,62 +81,102 @@ const isPrime = (num) => {
   // hint: a prime number is only evenly divisible by itself and 1
   // hint2: you can solve this using a for loop
   // note: 0 and 1 are NOT considered prime numbers
+
+  if (num === 0 || num === 1) {
+    return false;
+  }
+
+  for (let i = num - 1; i > 1; i--) {
+    if (num % i === 0) {
+      return false;
+    }
+  }
+
+  return true;
 };
 
 const returnFirst = (arr) => {
   // return the first item from the array
+  return arr.shift();
 };
 
 const returnLast = (arr) => {
   // return the last item of the array
+  return arr.pop();
 };
 
 const getArrayLength = (arr) => {
   // return the length of the array
+  return arr.length;
 };
 
 const incrementByOne = (arr) => {
   // arr is an array of integers
   // increase each integer by one
   // return the array
+  const incrementedArray = arr.map(element => element += 1);
+
+  return incrementedArray;
 };
 
 const addItemToArray = (arr, item) => {
   // add the item to the end of the array
   // return the array
+  arr.push(item);
```

Comment:
Opportunity for refactor - mutation. `push` mutates the parameter passed in and causes this method to have side effects. When working functionally, avoid all side-effects.

```javascript
const addItemToArray = (arr, item) => [...arr, item];
```